detect_secrets/core/secrets_collection.py

```diff
@@ -55,7 +55,8 @@ def scan_files(self, *filenames: str, num_processors: Optional[int] = None) -> N
         if not num_processors:
             num_processors = mp.cpu_count()
 
-        with mp.Pool(processes=num_processors) as p:
+        # Default to fork multiprocessing. Python 3.8+ defaults to spawn which doesn't share memory.
+        with mp.get_context('fork').Pool(processes=num_processors) as p:
```

Comment:
I did find on https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods

that it says `fork` should not be used on macOS, and that it's not available on Windows

so it may be better to resolve this some other way

Reply:
Updated to pass the settings object to the child processes.